sdk/typescript/src/cryptography/ed25519-keypair.ts

```diff
@@ -54,11 +54,21 @@ export class Ed25519Keypair implements Keypair {
   }
 
   /**
-   * Create a Ed25519 keypair from a raw secret key byte array.
-   *
-   * This method should only be used to recreate a keypair from a previously
-   * generated secret key.
+   * Create a Ed25519 keypair from a raw secret key byte array, also known as seed.
+   * This is NOT the private scalar which is resukt of hashing and bit clamping the
+   * secret key.
    *
+   * The sui.keystore key is a list of Base64 encoded `flag || privkey`. To import
+   * a key from sui.keystore to typescript, decode from base64 and remove the first
+   * flag byte after checking it is indeed the Ed25519 scheme flag 0x00:
+   * ```
+   * import { Ed25519Keypair, fromB64 } from '@mysten/sui.js';
+   * const raw = fromB64(t[1]);
+   * if (raw[0] !== 0 || raw.length !== 33) {
```

Comment:
ideally we should document that the flag  = 0 (either as a const or at least somewhere in the docs)

Reply:
added as comment here. technically this is a rust concept code, we dont use the flag byte anywhere in typescript.